vou/simulation.py

```diff
@@ -193,7 +203,40 @@ def update_availability(self, t: int):
     def record_dose_taken(self, t):
         """
         Takes the necessary actions when the person has taken a dose.
+
+        Increase record is used to update the dose source, type, and method
+        * If this is the first timestep, go with Primary and a random drug from that distribution
+        * If not, go with the last specified dose type and source --specified from will_increase_dose
+            * This will cause the type of dose to change each time increase is True
+        * Method of use will change each time a dose is taken based on the probabilities associated
+            with drug type
         """
+
+        # create dataframe of dose increases to determine last source
+        ## Currently using option 1 - last source is the continued source
+        if len(list(self.person.dose_increase_record.items())) > 0:
+            dose_dictionary = list(self.person.dose_increase_record.values())
+
+        ### Filter to the last entry where the source was not "WILL NOT INCREASE" and the attempt to increase was a success
+        ##### If that does not exist then the source/etc. will be based on primary
+            last_source = [x for x in dose_dictionary if (x['source'] != DoseIncreaseSource.WILL_NOT_INCREASE) & (x['success'] == True)]
+
+            # Determine source
+            if len(last_source) == 0: # if first record --primary
+                self.dose_source = DoseIncreaseSource.PRIMARY_DOCTOR
+                self.dose_type = weighted_random_by_dct(self.person.drug_params['drugs_by_source'][str(self.dose_source)])
+
+            else:
+                self.dose_source = last_source[-1]['source']
```

Comment:
This may be something to think through.  Potentially in another PR.  Right now the values would only be kept when changing (if I'm not mistaken).  We could extrapolate the rest, but depending on how the PK function needs to be incorporated could affect how we want to store the info.  I think in a pinch we can just add to a dictionary, but I'm open to thoughts based on how we want to incorporate this

Reply:
Good point, I agree that we can just pass the PK function whatever the current value of these attributes is. If we want to keep track of it for viz, we can add that later.